crates/oxc_language_server/src/capabilities.rs

```diff
@@ -0,0 +1,137 @@
+                    change_notifications: Some(OneOf::Left(true)),
+                }),
+                file_operations: None,
+            }),
+            code_action_provider: if value.code_action_provider {
+                Some(CodeActionProviderCapability::Options(CodeActionOptions {
+                    code_action_kinds: Some(vec![
+                        CodeActionKind::QUICKFIX,
+                        CODE_ACTION_KIND_SOURCE_FIX_ALL_OXC,
+                    ]),
+                    work_done_progress_options: WorkDoneProgressOptions {
+                        work_done_progress: None,
+                    },
+                    resolve_provider: None,
+                }))
+            } else {
+                None
+            },
+            ..ServerCapabilities::default()
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use tower_lsp::lsp_types::{
+        ClientCapabilities, CodeActionClientCapabilities, CodeActionKindLiteralSupport,
+        CodeActionLiteralSupport, TextDocumentClientCapabilities,
+    };
+
+    use super::Capabilities;
+
+    /// this is from build (see help, about):
+    /// Version: 1.95.3 (user setup)
+    /// Commit: f1a4fb101478ce6ec82fe9627c43efbf9e98c813
+    /// Date: 2024-11-13T14:50:04.152Z
+    /// Electron: 32.2.1
+    /// ElectronBuildId: 10427718
+    /// Chromium: 128.0.6613.186
+    /// Node.js: 20.18.0
+    /// V8: 12.8.374.38-electron.0
+    /// OS: Windows_NT x64 10.0.19045
+    #[test]
+    fn test_code_action_provider_vscode() {
+        let client_capabilities = ClientCapabilities {
+            text_document: Some(TextDocumentClientCapabilities {
+                code_action: Some(CodeActionClientCapabilities {
+                    code_action_literal_support: Some(CodeActionLiteralSupport {
+                        code_action_kind: CodeActionKindLiteralSupport {
+                            value_set: vec![
+                                #[allow(clippy::manual_string_new)]
+                                "".into(),
+                                "quickfix".into(),
+                                "refactor".into(),
+                                "refactor.extract".into(),
+                                "refactor.inline".into(),
+                                "refactor.rewrite".into(),
+                                "source".into(),
+                                "source.organizeImports".into(),
+                            ],
+                        },
+                    }),
+                    ..CodeActionClientCapabilities::default()
+                }),
+                ..TextDocumentClientCapabilities::default()
+            }),
+            ..ClientCapabilities::default()
+        };
+
+        let capabilities = Capabilities::from(client_capabilities);
+
+        assert!(capabilities.code_action_provider);
+    }
+
+    #[test]
+    fn test_code_action_provider_intellij() {
+        let client_capabilities = ClientCapabilities {
```

Comment:
Full IntelliJ info below.  I think my comments on the VS Code test apply here as well, but I wanted to provide you all the information in case you felt differently.

<summary>
<details>
<pre>
IntelliJ IDEA 2024.3.1 (Ultimate Edition)
Build #IU-243.22562.145, built on December 8, 2024
Licensed to John Smith
Subscription is active until July 23, 2025.
Runtime version: 21.0.5+8-b631.28 aarch64 (JCEF 122.1.9)
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Toolkit: sun.lwawt.macosx.LWCToolkit
macOS 15.2
Kotlin analyzer version: 2.1.20-ij243-27
GC: G1 Young Generation, G1 Concurrent GC, G1 Old Generation
Memory: 2048M
Cores: 10
Metal Rendering is ON
Registry:
  ide.experimental.ui=true
  i18n.locale=
Non-Bundled Plugins:
  com.github.oxc.project.oxcintellijplugin (0.0.1)
Kotlin: 243.22562.145-IJ
</pre>
<pre>
ClientCapabilities [
  workspace = WorkspaceClientCapabilities [
    applyEdit = true
    workspaceEdit = WorkspaceEditCapabilities [
      documentChanges = true
      resourceOperations = SingletonList (
        "create"
      )
      failureHandling = "abort"
      normalizesLineEndings = true
      changeAnnotationSupport = null
    ]
    didChangeConfiguration = null
    didChangeWatchedFiles = DidChangeWatchedFilesCapabilities [
      relativePatternSupport = true
      dynamicRegistration = true
    ]
    symbol = null
    executeCommand = ExecuteCommandCapabilities [
      dynamicRegistration = false
    ]
    workspaceFolders = true
    configuration = null
    semanticTokens = SemanticTokensWorkspaceCapabilities [
      refreshSupport = true
    ]
    codeLens = null
    fileOperations = null
    inlayHint = null
    inlineValue = null
    diagnostics = null
  ]
  textDocument = TextDocumentClientCapabilities [
    synchronization = SynchronizationCapabilities [
      willSave = false
      willSaveWaitUntil = false
      didSave = true
      dynamicRegistration = false
    ]
    completion = CompletionCapabilities [
      completionItem = CompletionItemCapabilities [
        snippetSupport = true
        commitCharactersSupport = null
        documentationFormat = ArrayList (
          "markdown",
          "plaintext"
        )
        deprecatedSupport = true
        preselectSupport = null
        tagSupport = CompletionItemTagSupportCapabilities [
          valueSet = SingletonList (
            Deprecated
          )
        ]
        insertReplaceSupport = true
        resolveSupport = CompletionItemResolveSupportCapabilities [
          properties = SingletonList (
            "documentation"
          )
        ]
        insertTextModeSupport = null
        labelDetailsSupport = true
      ]
      completionItemKind = CompletionItemKindCapabilities [
        valueSet = EnumEntriesList (
          Text,
          Method,
          Function,
          Constructor,
          Field,
          Variable,
          Class,
          Interface,
          Module,
          Property,
          Unit,
          Value,
          Enum,
          Keyword,
          Snippet,
          Color,
          File,
          Reference,
          Folder,
          EnumMember,
          Constant,
          Struct,
          Event,
          Operator,
          TypeParameter
        )
      ]
      contextSupport = null
      insertTextMode = null
      completionList = CompletionListCapabilities [
        itemDefaults = ArrayList (
          "commitCharacters",
          "editRange",
          "insertTextFormat",
          "insertTextMode",
          "data"
        )
      ]
      dynamicRegistration = null
    ]
    hover = HoverCapabilities [
      contentFormat = ArrayList (
        "markdown",
        "plaintext"
      )
      dynamicRegistration = null
    ]
    signatureHelp = null
    references = ReferencesCapabilities [
      dynamicRegistration = true
    ]
    documentHighlight = null
    documentSymbol = null
    formatting = FormattingCapabilities [
      dynamicRegistration = true
    ]
    rangeFormatting = null
    onTypeFormatting = null
    declaration = null
    definition = DefinitionCapabilities [
      linkSupport = true
      dynamicRegistration = null
    ]
    typeDefinition = TypeDefinitionCapabilities [
      linkSupport = true
      dynamicRegistration = null
    ]
    implementation = null
    codeAction = CodeActionCapabilities [
      codeActionLiteralSupport = CodeActionLiteralSupportCapabilities [
        codeActionKind = CodeActionKindCapabilities [
          valueSet = ArrayList (
            "quickfix",
            "",
            "source",
            "refactor"
          )
        ]
      ]
      isPreferredSupport = null
      disabledSupport = true
      dataSupport = true
      resolveSupport = CodeActionResolveSupportCapabilities [
        properties = SingletonList (
          "edit"
        )
      ]
      honorsChangeAnnotations = null
      dynamicRegistration = null
    ]
    codeLens = null
    documentLink = null
    colorProvider = ColorProviderCapabilities [
      dynamicRegistration = true
    ]
    rename = null
    publishDiagnostics = PublishDiagnosticsCapabilities [
      relatedInformation = null
      tagSupport = Either [
        left = null
        right = DiagnosticsTagSupport [
        valueSet = ArrayList (
          Unnecessary,
          Deprecated
        )
      ]
      ]
      versionSupport = true
      codeDescriptionSupport = null
      dataSupport = null
    ]
    foldingRange = null
    typeHierarchy = null
    callHierarchy = null
    selectionRange = null
    semanticTokens = SemanticTokensCapabilities [
      requests = SemanticTokensClientCapabilitiesRequests [
        range = Either [
          left = false
          right = null
        ]
        full = Either [
          left = null
          right = SemanticTokensClientCapabilitiesRequestsFull [
          delta = false
        ]
        ]
      ]
      tokenTypes = ArrayList (
        "namespace",
        "type",
        "class",
        "enum",
        "interface",
        "struct",
        "typeParameter",
        "parameter",
        "variable",
        "property",
        "enumMember",
        "event",
        "function",
        "method",
        "macro",
        "keyword",
        "modifier",
        "comment",
        "string",
        "number",
        "regexp",
        "operator",
        "decorator"
      )
      tokenModifiers = ArrayList (
        "declaration",
        "definition",
        "readonly",
        "static",
        "deprecated",
        "abstract",
        "async",
        "modification",
        "documentation",
        "defaultLibrary"
      )
      formats = SingletonList (
        "relative"
      )
      overlappingTokenSupport = true
      multilineTokenSupport = true
      serverCancelSupport = false
      augmentsSyntaxTokens = null
      dynamicRegistration = null
    ]
    moniker = null
    linkedEditingRange = null
    inlayHint = null
    inlineValue = null
    diagnostic = null
  ]
  notebookDocument = null
  window = WindowClientCapabilities [
    workDoneProgress = null
    showMessage = WindowShowMessageRequestCapabilities [
      messageActionItem = null
    ]
    showDocument = ShowDocumentCapabilities [
      support = true
    ]
  ]
  general = GeneralClientCapabilities [
    regularExpressions = null
    markdown = null
    staleRequestSupport = StaleRequestCapabilities [
      cancel = true
      retryOnContentModified = ArrayList ()
    ]
    positionEncodings = null
  ]
  experimental = null
]
</pre>
</details>
</summary>